### PR TITLE
Remove git subtree and update to 5.34. JB#56962

### PR DESCRIPTION
**Pull request by pvuorela**
Got rid of the painful git subtree, removed the extra sources and then bumped to 5.34 which was still relatively simple. Separate commit for each step.

Going forward it would be good to first get rid of bluez4 support altogether which is included here as relatively big patch. Also for some reason the upstream cmake has been replaced with custom qmake.

@Tomin1 @spiiroin

### CODE REVIEW

**Comment by pvuorela** [comment]
FWIW for ensuring the git subtree removal is good, after the first commit the sources can be applied and then the old source copy directory should match what the upstream directory has.

**Comment by spiiroin** [comment]
This is a bit hard to parse (adding more meat to commit messages would not hurt). Am I right to assume it goes something like below?

In starting point there is:

- git submodule "upstream" - which is sort of for reference, but not used in build
- local git subdirectory "bluez-qt" that contains some version of upstream + local commits
- build is done from sources in "bluez-qt"

First commit then:
- makes patches out of local commits in "blues-qt" subdir
- these  patches are applied to "upstream" submodule content
- sources from "upstream" submodule are used during build
- so that "blues-qt" subdirectory is no longer used in build

Second commit:

- removes the now unused "blues-qt" subdirectory
- we have a more usual upstream as-it-is + patches type situation

And the last commit:

- brings the "upstream" version in use more up to date
- updates patch set accordingly

**Comment by pvuorela** [comment]
You got it right. This was still using the "git subtree" structure that was in fashion on our repositories around 2016 when this repository was set up. Since removed on most places already.

File: rpm/0001-Add-rpm-spec-and-.pro-files.patch [hunk 1]
```diff
@@ -1,7 +1,7 @@
 From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
 From: Bea Lam <bea.lam@jollamobile.com>
 Date: Thu, 21 Jul 2016 05:37:23 +0930
-Subject: [PATCH] [packaging] Add rpm spec and .pro files
+Subject: [PATCH] Add rpm spec and .pro files
```
Comment (spiiroin) [inline rpm/0001-Add-rpm-spec-and-.pro-files.patch:4]: While the original commit added spec too, the patch does not -> subject could be adjusted?

**Review by spiiroin — APPROVED** [review]

File: rpm/0001-Add-rpm-spec-and-.pro-files.patch (hunk 1 above)
Comment (pvuorela) [reply rpm/0001-Add-rpm-spec-and-.pro-files.patch:4]: Was about to adjust manually on the patch but then realized it would the next time create a patch file with different name with git format-patch. Since we should get rid of the whole .pro files on hopefully the next update, perhaps I just just leave it as it is now.

Comment (Thaodan) [reply rpm/0001-Add-rpm-spec-and-.pro-files.patch:4]: You can keep the tag by using git format-patch -k/--keep-subject

Comment (pvuorela) [reply rpm/0001-Add-rpm-spec-and-.pro-files.patch:4]: " Do not strip/add [PATCH] from the first line of the commit log message." - that doesn't help here much. Point was that "rpm spec" should be removed from the subject and if I did that manually, git had no way to know to generate such words when creating patches the next time.
